docker/docker-compile.sh

```diff
@@ -0,0 +1,218 @@
+    cp "${project_dir}"/etc/vconsole.conf "${project_dir}"/etc/locale.gen "${squashfs}"/etc/
+    arch-chroot "${squashfs}" /bin/bash locale-gen
+
+    # Copy over main Anarchy config and installer script, make them executable
+    echo "Adding anarchy config and installer scripts to iso ..."
+    cp "${project_dir}"/etc/anarchy.conf "${project_dir}"/etc/pacman.conf "${squashfs}"/etc/
+    cp "${project_dir}"/anarchy-installer.sh "${squashfs}"/usr/bin/anarchy
+    cp "${project_dir}"/extra/sysinfo "${project_dir}"/extra/iptest "${squashfs}"/usr/bin/
+    chmod +x "${squashfs}"/usr/bin/anarchy "${squashfs}"/usr/bin/sysinfo "${squashfs}"/usr/bin/iptest
+
+    # Create Anarchy and lang directories, copy over all lang files
+    echo "Adding language files to iso ..."
+    mkdir -p "${squashfs}"/usr/share/anarchy/lang "${squashfs}"/usr/share/anarchy/extra "${squashfs}"/usr/share/anarchy/boot "${squashfs}"/usr/share/anarchy/etc
+    cp "${project_dir}"/lang/* "${squashfs}"/usr/share/anarchy/lang/
+
+    # Create shell function library, copy /lib to squashfs-root
+    echo "Adding anarchy scripts to iso ..."
+    mkdir "${squashfs}"/usr/lib/anarchy
+    cp "${project_dir}"/lib/* "${squashfs}"/usr/lib/anarchy/
+
+    # Copy over extra files (dot files, desktop configurations, help file, issue file, hostname file)
+    echo -e "Adding dot files and desktop configurations to iso ..."
+    rm "${squashfs}"/root/install.txt
+    cp "${project_dir}"/extra/shellrc/.zshrc "${squashfs}"/root/
+    cp "${project_dir}"/extra/.help "${project_dir}"/extra/.dialogrc "${squashfs}"/root/
+    cp "${project_dir}"/extra/shellrc/.zshrc "${squashfs}"/etc/zsh/zshrc
+    cp -r "${project_dir}"/extra/shellrc/. "${squashfs}"/usr/share/anarchy/extra/
+    cp -r "${project_dir}"/extra/desktop "${project_dir}"/extra/fonts "${project_dir}"/extra/anarchy-icon.png "${squashfs}"/usr/share/anarchy/extra/
+    cat "${project_dir}"/extra/.helprc | tee -a "${squashfs}"/root/.zshrc >/dev/null
+    cp "${project_dir}"/etc/hostname "${project_dir}"/etc/issue_cli "${squashfs}"/etc/
+    cp -r "${project_dir}"/boot/splash.png "${project_dir}"/boot/loader/ "${squashfs}"/usr/share/anarchy/boot/
+    cp "${project_dir}"/etc/nvidia340.xx "${squashfs}"/usr/share/anarchy/etc/
+
+    if [ -d branding ]; then
+        cp "${project_dir}"/branding/wallpapers/* "${squashfs}"/usr/share/anarchy/extra/wallpapers/
+    else
+        echo "Missing branding directory, skipping ..."
+    fi
+
+    # Copy over built packages and create repository
+    echo "Adding built AUR packages to iso ..."
+    mkdir "${custom_iso}"/arch/x86_64/squashfs-root/usr/share/anarchy/pkg
+
+    for pkg in $(echo "${local_aur_packages[@]}"); do
+        cp /tmp/"${pkg}"/*.pkg.tar.xz "${squashfs}"/usr/share/anarchy/pkg/
+    done
+
+    cd "${squashfs}"/usr/share/anarchy/pkg || exit
+    repo-add anarchy-local.db.tar.gz *.pkg.tar.xz
```

Comment:
The compiled packages don't seem to be copied over to the decompiled iso, so the installation fails every time, since it does not find the local repo.

Reply:
Sorry. Fixed. 
Strangely `makepkg` inside docker builds packages into `.tar`  files so manually converted them to `tar.xz` (docker/setup.sh:74).

Reply:
No problem, but it seems the issue is still present.
![Screenshot from 2020-03-26 13-19-50](https://user-images.githubusercontent.com/46302511/77646325-9d9dd200-6f64-11ea-9e8c-429e72b77999.png)

Reply:
I got this same error using Qemu as VM while ISO worked fine on Hyper-V. I thought it was due to lack of resources. I was able to install a full desktop using generated ISO. Are you testing by Qemu?
I do the development part in an Arch linux inside a VM on Hyper-V and uploaded generated ISO so I was able to download it in host OS and setup a new VM in Hyper-V (VirtualBox doesn't work inside such a VM).

Reply:
Yes, I'm using QEMU to test the iso, but I don't understand why this would change the fact that the files weren't properly copied.

I'll try allocating it more memory (I tested with 2 GB) if this was the problem, but I'm also using KVM (`-enable-kvm` in QEMU), so resources really shouldn't be the problem, especially since this worked fine with the same settings before your commits.

I'll try generating the iso on another laptop, but if allocating more memory in QEMU doesn't work, I can't accept the PR until the issue is fixed (or we figure out a workaround), since we can't know for sure what method to generate the iso and/or install Anarchy our users will use.

Reply:
Weird, this just happened and it appears I didn't notice before, when compiling the iso:

![Screenshot from 2020-03-26 13-43-59](https://user-images.githubusercontent.com/46302511/77648152-f884f880-6f67-11ea-8c13-2ea2e81d299a.png)

Reply:
Meanwhile I will find a working Qemu config. I've tried KVM inside that Arch VM but it failed (error message said it is not available/installed something like that).

> I can't accept the PR until the issue is fixed...

Sure

Reply:
I think the issue might be that the docker script has to clean up after itself (by removing the image).

I've run the script before you implemented your changes, so it didn't work the first time, but I re-ran the script after your commits and it appears it didn't recompile the packages since it found the existing image. So if you add something to the script to clean up existing images at the end I think it might fix the problem.

Reply:
You need to remove `anarchy:latest` manually before testing new version:

`docker rmi anarchy:latest`

`setup.sh` is hard-coded inside the image. Rest are dynamic.

Reply:
I'll test it now, but can you replace the "[Found local docker image](https://github.com/AnarchyLinux/installer/blob/de53e4301144432536c574dbfd3adf5b46cf7c5c/compile.sh#L353)" line with the `docker rmi anarchy:latest` command (and possibly some text explaining the image was found and is being removed), to make sure every compilation is completely clean/"fresh".

Reply:
I test once with a clean environment (changes were simple and reasonable) now I'm trying to test the generated ISO with same docker image (it should work the same unless docker image was made with a different `setup.sh`).
If using the same image makes the problem I'll come up with fix. Generating docker image on each compile is a lot of unnecessary work.
I based the changes on master. Should I use log?

Reply:
> If using the same image makes the problem I'll come up with fix. Generating docker image on each compile is a lot of unnecessary work.

Although I agree (and it does take a lot more time to generate the iso), I still think this would be best, since we can make sure the generated iso is totally clean and made from the latest commits, since currently (excluding your docker branch), the script cleans up residual files and recompiles all the packages. Compilations are fairly rare, so I think it's worth the extra time if it means we don't have to worry about manually removing the docker image.

> I based the changes on master. Should I use log?

Sorry, apparently I got some commits mixed up (since they're in both branches). No need to rebase, master was the preferred branch anyway.

Reply:
Your recommendation to remove the docker image worked, so I think the automatic removal is the last thing issue and then we can merge the PR. Do you mind if I squash the commits when merging?

Reply:
I'll make it (removing docker image) default for now and come up with the fix (if any needed) and make it optional soon.

> Do you mind if I squash the commits when merging

Anything is OK about my covidribution.

Reply:
Everything works fine now, thank you very much for the contribution.

Since there were only 5 commits, all for the same feature, I'll squash and merge, to keep the branch clean.